### PR TITLE
Copy container

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* Created an API to copy container from rest endpoint where path parameter id must be provided as an identifier of the source container. Id parameter must also qualify the uuid format

### Checklist
- [X] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

https://venturedive-plantiga.slack.com/files/U03TTHG6L6R/F03VDE1P84B/image.png

https://files.slack.com/files-pri/T0255NE5ZQ8-F03VAJ5UR8V/image.png

https://files.slack.com/files-pri/T0255NE5ZQ8-F0404SRKUFL/image.png

https://files.slack.com/files-pri/T0255NE5ZQ8-F040HFNTANM/image.png

https://files.slack.com/files-pri/T0255NE5ZQ8-F0404SV6ZQA/image.png

### CODE REVIEW

File: dotCMS/dotsecure/license/server_id.dat [hunk 1]
```diff
@@ -0,0 +1 @@
+ba9e1810-8816-434a-bb3a-c880b4581db1
```
Comment (jcastro-dotcms) [inline dotCMS/dotsecure/license/server_id.dat:1]: Exclude this file from the commit please.

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/dotsecure/license/server_id.dat (hunk 1 above)
Comment (jdotcms) [inline dotCMS/dotsecure/license/server_id.dat:1]: this should be not part of the PR

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java [hunk 2]
```diff
@@ -1068,4 +1061,65 @@ public final Response delete(@Context final HttpServletRequest  request,
...
+     * Copies container to the specified host
+     *
+     * @param request
+     * @param response
+     * @param id       id identifier to copy.
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @POST
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1074]: We need to add the Swagger annotations

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java [hunk 3]
```diff
@@ -1068,4 +1061,65 @@ public final Response delete(@Context final HttpServletRequest  request,
...
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).requiredBackendUser(true)
+                .rejectWhenNoUser(true).init();
+        final User user = initData.getUser();
+        final Host host = WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request);
+        final PageMode pageMode = PageMode.get(request);
+
+        if (!UtilMethods.isSet(id)) {
+
+            Logger.error(this, "The container id is required");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1093]: We used to throw an exception rather than return the Response

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java [hunk 4]
```diff
@@ -1068,4 +1061,65 @@ public final Response delete(@Context final HttpServletRequest  request,
...
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @POST
+    @Path("/{id}/_copy")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response copy(@Context final HttpServletRequest request,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1080]: it would be interesting here to think if the user would like or not add a new name in advance

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java [hunk 5]
```diff
@@ -1068,4 +1061,65 @@ public final Response delete(@Context final HttpServletRequest  request,
...
+                    "User " + user.getPrimaryKey() + " saved " + sourceContainer.getTitle(),
+                    host.getHostname());
+
+            Container copiedContainer = this.containerAPI.copy(sourceContainer, host, user,
+                    pageMode.respectAnonPerms);
+
+            Logger.debug(this,
+                    () -> "The container: " + sourceContainer.getIdentifier() + " has been copied");
+
+            return Response.ok(new ResponseEntityView(new ContainerView(copiedContainer))).build();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1118]: As part of the effort to doc new things with swagger, we create an implementation of the ResponseEntityView
that you can return it (without wrapping it on the Response) or you can wrap into the Response but need to extend the ResponseEntityView and type it for instance:
```
ContainerViewResponseEntityView extends ResponseEntityView<ContainerView>
```

This will give enough information to swagger to create a doc for the response

File: dotCMS/dotsecure/license/server_id.dat (hunk 1 above)
Comment (hassan-mustafa-baig) [reply dotCMS/dotsecure/license/server_id.dat]: File has been removed. It was basically automatically added when I build through gradle task and it was skipped during commit

Comment (hassan-mustafa-baig) [reply dotCMS/dotsecure/license/server_id.dat]: I have addressed this in my last commit

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java (hunk 4 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1080]: This was discussed with @freddyucv and @dsilvam and it was first confirmed then implemented

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java (hunk 2 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1075]: I have updated the code as per discussion with @dsilvam so @jdotcms can you please review the last commit and share your feedback please

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java (hunk 3 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java:1093]: Code has been updated according to the feedback

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java (hunk 5 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java]: Kindly review the recent commit - thanks

**Review by jdotcms — APPROVED** [review]
